### PR TITLE
jxl-render: Render with crop regions

### PR DESCRIPTION
**Pull request by tirr-c**
TODO:
- [x] Wrong HF coeffs are used when topleft corner of the region is not on the first group

### CODE REVIEW

File: crates/jxl-render/src/features/spline.rs [hunk 1]
```diff
@@ -261,12 +264,20 @@ pub fn render_spline(
...
+                    }
+
                     for x in xbegin..xend {
-                        let sample = buffer.get_mut(x as usize, y as usize).unwrap();
+                        let fx = x - region.left;
+                        if fx < 0 {
+                            continue;
+                        }
+
+                        let sample = buffer.get_mut(fx as usize, fy as usize).unwrap();
```
Comment (EugeneVert) [inline crates/jxl-render/src/features/spline.rs:280]:
```log
'main' panicked at 'called `Option::unwrap()` on a `None` value', crates/jxl-render/src/features/spline.rs:280:79
```
With `--crop` on images with splines

Example: jxl-dec spline jxl --crop "128 128 10 10"

Comment (tirr-c) [reply crates/jxl-render/src/features/spline.rs:280]: Oops, I'll take a look

**Review by EugeneVert — APPROVED** [review]
Great work!

File: crates/jxl-render/src/features/spline.rs (hunk 1 above)
Comment (tirr-c) [reply crates/jxl-render/src/features/spline.rs]: It was caused by checking only left bounds, but not right bounds. It is now fixed.

Comment (EugeneVert) [reply crates/jxl-render/src/features/spline.rs]: Thanks!

**Pull request merged**